### PR TITLE
Feat/update header mobile sections

### PR DESCRIPTION
**Pull request by houssemmimoun27**
This PR implement the new header mobile menu.

Screenshot 2023-04-27 at 22 50 35
 
Screenshot 2023-04-27 at 22 50 17

### CODE REVIEW

File: next.config.js [hunk 1]
```diff
@@ -21,6 +21,42 @@ module.exports = {
       }
     ]
   },
+  async rewrites() {
```
Comment (ccamel) [inline next.config.js:24]: What's the purpose of rewriting the URLs?

Comment (houssemmimoun27) [reply next.config.js:24]: The goal here is to link the `FAQ`, `Terms` and `Rewards` pages to the same path under the pages folder as they are the same under the folders validators and builders program but the text content will be changed via the router path. So in order to not recreate them under the two folders, i linked them with rewriting the URLs.

File: next.config.js [hunk 2]
```diff
@@ -21,6 +21,42 @@ module.exports = {
...
+    return [
+      {
+        source: '/validators',
+        destination: '/'
+      },
+      {
+        source: '/builders',
+        destination: '/'
+      },
```
Comment (fredericvilcot) [inline next.config.js:33]: OK, but do we want to have dedicated `/validators` and `/builders` routes ? 

They have no entry point, they are useless.

IMHO, in the header, when you click on `Validators` or `Builders`, it should not push a new route in history, but only open the submenu as side effect.

File: src/routes.ts [hunk 3]
```diff
@@ -1,43 +1,85 @@
...
+  name: string
+  path: string
+  hash?: string
+}
+
 export type Route = {
   name: string
   path: string
   hash?: string
+  title: string
```
Comment (fredericvilcot) [inline src/routes.ts:11]: What is the purpose of the `title` property? 🧐

You already have the `name` property that is sufficient to describe the route.

File: src/routes.ts [hunk 4]
```diff
@@ -1,43 +1,85 @@
+export type SubRoute = {
+  name: string
+  path: string
+  hash?: string
+}
```
Comment (fredericvilcot) [inline src/routes.ts:5]:
```suggestion
export type SubRoute = Omit<Route, 'menu'>
```

File: src/routes.ts [hunk 5]
```diff
@@ -1,43 +1,85 @@
...
 export type Route = {
   name: string
   path: string
   hash?: string
+  title: string
+  menu?: SubRoute[]
 }
 
-export const headerRoutes: Route[] = [
+const subMenuLinks = (link: string): SubRoute[] => [
```
Comment (fredericvilcot) [inline src/routes.ts:15]:
```suggestion
const subMenuLinks = (path: string): SubRoute[] => [
```

File: src/components/layout/header/Header.tsx [hunk 6]
```diff
@@ -94,51 +85,74 @@ const DesktopMenu: React.FC<MenuProps> = ({ routerPath }): JSX.Element => {
...
+                separatorElement}
+            </div>
+          ))}
+        </div>
       </div>
-      <SubMenuDesktop
-        routerPath={routerPath}
-        selectedMenu={selectedMenu}
-      />
+      {!isMobileScreen && <SubMenuDesktop routerPath={routerPath} selectedMenu={selectedMenu} />}
```
Comment (fredericvilcot) [inline src/components/layout/header/Header.tsx:148]: This could be simplified with one single ternary to prevent from multiple assertions.

**Review by fredericvilcot — COMMENTED** [review]
Great, just a minor enhancement 😌

**Review by fredericvilcot — APPROVED** [review]

**Pull request merged**